ProcessLib/Process.cpp

```diff
@@ -168,6 +172,48 @@ void Process::constructDofTable()
         std::move(all_mesh_subsets), NumLib::ComponentOrder::BY_LOCATION));
 }
 
+void Process::initializeExtrapolator()
+{
+    NumLib::LocalToGlobalIndexMap const* dof_table_single_component;
+    bool manage_storage;
+
+    if (_local_to_global_index_map->getNumberOfComponents() == 1)
+    {
+        // For single-variable-single-component processes reuse the existing DOF
+        // table.
+        dof_table_single_component = _local_to_global_index_map.get();
+        manage_storage = false;
+    }
+    else
+    {
+        // Otherwise construct a new DOF table.
+        std::vector<std::unique_ptr<MeshLib::MeshSubsets>>
```

Comment:
why do you need to create a new DoF table here? Isn't it possible to somehow reuse the existing one?